src/pages/index.js

```diff
@@ -53,45 +63,220 @@ const addFormValidator = new FormValidator(
+    });
+}
+
+function handleDeleteConfirm(data) {
+  const cardId = data.cardId;
+  if (!cardId || typeof cardId !== "string") {
+    console.error(
+      "Invalid cardId in handleDeleteConfirm:",
+      cardId,
+      "Data:",
+      data
+    );
+    return;
+  }
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      section.removeItem(cardId);
+      deleteCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+    });
+}
+
+function handleAvatarFormSubmit(data) {
+  avatarSubmitButton.textContent = "Saving...";
+  avatarSubmitButton.classList.add("loading");
+  avatarSubmitButton.disabled = true;
+  api
+    .updateUserAvatar(data.avatar)
+    .then((updatedInfo) => {
+      userInfo.setUserInfo({
+        name: updatedInfo.name,
+        about: updatedInfo.about,
+        avatar: updatedInfo.avatar,
+      });
+      avatarEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar", err);
+    })
+    .finally(() => {
+      avatarSubmitButton.textContent = "Save";
+      avatarSubmitButton.classList.remove("loading");
+      avatarSubmitButton.disabled = false;
+    });
 }
 
+avatarEditButton.addEventListener("click", () => {
+  const currentUserInfo = userInfo.getUserInfo();
+  avatarInput.value = currentUserInfo.avatar || "";
+  avatarEditPopup.open();
+  avatarFormValidator.resetValidation();
+  avatarFormValidator.enableValidation();
+});
+
+const imagePopup = new PopupWithImage("#preview-modal");
+
+const profileEditPopup = new PopupWithForm(
+  "#profile__edit-modal",
+  handleProfileFormSubmit
+);
+
+const addCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
+);
+
+const deleteCardPopup = new PopupWithConfirm(
+  "#confirm__delete-modal",
+  handleDeleteConfirm
+);
+
+const avatarEditPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleAvatarFormSubmit
+);
+
+const section = new Section(
+  { items: [], renderer: (cardData) => createCard(cardData) },
+  ".cards__list"
+);
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    section.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error("Error fetching cards:", err);
+  });
+
+api
+  .getUserInfo()
+  .then((info) => {
+    userInfo.setUserInfo({
+      name: info.name,
+      about: info.about,
+      avatar: info.avatar,
+    });
+  })
+  .catch((err) => {
+    console.error("Error fetching user info:", err);
+  });
+
+avatarEditPopup.setEventListeners();
+profileEditPopup.setEventListeners();
+addCardPopup.setEventListeners();
+deleteCardPopup.setEventListeners();
+imagePopup.setEventListeners();
+
 profileEditButton.addEventListener("click", () => {
   const currentUserInfo = userInfo.getUserInfo();
   profileTitleInput.value = currentUserInfo.name;
-  profileDescriptionInput.value = currentUserInfo.job;
+  profileDescriptionInput.value = currentUserInfo.about;
   editFormValidator.resetValidation();
+  editFormValidator.enableValidation();
```

Comment:
`enableValidation` should be called only 1 time to enable validation. You should not do that when you open/close a popup because you add more and more the same listeners to the same forms. This may cause a `memory leak`